caching.py

```
@@ -0,0 +1,613 @@
+            
+            # Execute function and cache result
+            result = func(*args, **kwargs)
+            cache_manager.multi_tier_cache.set(key, result, ttl or cache_manager.default_ttl)
+            
+            return result
+        
+        return wrapper
+    return decorator
+
+
+def with_cache(shared: Dict[str, Any], ttl: Optional[float] = None):
```

Ruff (0.12.2)

Missing return type annotation for public function `with_cache`

(ANN201)